src/file_dialog.rs

```diff
@@ -3069,3 +3110,46 @@ impl FileDialog {
         }
     }
 }
+
+fn truncate_filenames(path: &str, max_length: usize, is_dir: bool) -> String {
+    if path.len() <= max_length {
+        return path.to_string();
+    }
+
+    let path = Path::new(path);
+    let file_name = path.file_name().and_then(|f| f.to_str()).unwrap_or("");
+    let extension = if is_dir {
+        ""
+    } else {
+        path.extension().and_then(|ext| ext.to_str()).unwrap_or("")
+    };
+    let file_stem = file_name
+        .strip_suffix(&format!(".{extension}"))
+        .unwrap_or(file_name);
```

Comment:
`std::path` already provides a `file_stem` method: https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem

Could you use that instead?

Reply:
Yeah, I'll look into that, I might simplify it even further.